axlearn/cloud/gcp/job.py

```diff
@@ -563,23 +563,29 @@ def _build_pod(self) -> Nested[Any]:
             }
         )
 
+        pod_spec = dict(
+            terminationGracePeriodSeconds=60,
+            # Fail if any pod fails, and allow retries to happen at JobSet level.
+            restartPolicy="Never",
+            nodeSelector={
+                "cloud.google.com/gke-tpu-accelerator": system.gke_accelerator,
+                "cloud.google.com/gke-tpu-topology": system.topology,
+                **selector,
+            },
+            tolerations=tolerations,
+            containers=[self._build_container()],
+            serviceAccountName=cfg.service_account,
+            volumes=volumes,
+        )
+
+        # hostNetwork True and dnsPolicy do not work with Workload Identity and GCS Fuse.
```

Comment:
I suppose we should check here that workload identity is not being used, not just gcs fuse --- as sync'ed offline, will do more testing to see whether necessary before merging.

Reply:
GKE Workload Identity requires forcing metadata traffic to a specific pod. However, if you set hostNetwork true that's no longer possible so it doesn't work.

Note your GKE clusters can keep on using Workload Identity, however the issue is that it won't work for any pod that is using hostNetwork: true. All your other pods using hostNetwork: false will continue to be able to utilize workload identity just like before.

Reply:
We can attach the right service account (the same one used in the container) to GKE node pools instead of using the default one. Then it should work with hostNetwork=True